api/src/models/mine.js

```diff
@@ -0,0 +1,26 @@
+let mongoose = require('mongoose');
+let Mixed = mongoose.Schema.Types.Mixed;
+
+module.exports = require('../utils/model-schema-generator')(
+  'Mine',
+  {
+    _schemaName: { type: String, default: 'Mine', index: true },
+    name: { type: String, default: '' },
+    permitNumber: { type: String, default: '' },
+    status: { type: String, default: '' },
+    type: { type: String, default: '' },
+    commodities: { type: String, default: '' },
+    // number of tailings impoundments
+    tailingsImpoundments: { type: Number, default: 0 },
+    region: { type: String, default: '' },
+    centroid : [{ type: Mixed, default: 0.00}],
```

Comment:
I think keeping the centroid in the model is good. Maybe rename it from centroid to "Location" (a centroid implies it's the centre of a polygon feature... points are by definition always their centroid).

This way, we can use mongo's geojson query options (in fact, maybe the type should be a geojson point type rather than  coord), and for anything that needs long/lat, we can just derive dynamic columns from the spatial: location.geometry.coordinates[0/1];

Reply:
Great idea. I am not sure what the other models look like, but I was a bit leery of centroid as I knew it was matching the definition.

Reply:
Right yeah, I was going off of EPIC here. Had a quick chat in discord about this, and it was brought up that The reason EPIC used it this way is that mongo expects spacial data to be structured like this. The eventual conclusion was that this should probably be called location to be more descriptive and should be a GeoJSON object.